src/NServiceBus.Transport.RabbitMQ.Tests/ApprovalFiles/APIApprovals.Approve.approved.txt

```diff
@@ -5,44 +5,58 @@ namespace NServiceBus
     public class RabbitMQTransport : NServiceBus.Transport.TransportDefinition
     {
         public RabbitMQTransport() { }
-        public override string ExampleConnectionStringForErrorMessage { get; }
-        public override NServiceBus.Transport.TransportInfrastructure Initialize(NServiceBus.Settings.SettingsHolder settings, string connectionString) { }
+        public RabbitMQTransport(string connectionString) { }
+        public System.Security.Cryptography.X509Certificates.X509Certificate2 ClientCertificate { get; set; }
+        public System.TimeSpan HeartbeatInterval { get; set; }
+        public string Host { get; set; }
```

Comment:
Should we allow setting `Host` once the `ConnetionString` is set? 

Same question for other parts of the connection string.

Reply:
I treat connection string as a way to pass initial bunch of parameters that you can tweak later. I think this is in line with the API shape because the property-based design allows you to set and re-set things as many times as you want.